From d7b8c1eeb6c36800a891584ad948b4b5ab1f52bf Mon Sep 17 00:00:00 2001
From: armando-migliaccio <amigliaccio@nicira.com>
Date: Mon, 3 Jun 2013 13:57:44 -0700
Subject: [PATCH] Fix bug in port creation form

'network_name' is not a recognized Quantum attribute, so passing it
in the request to the Quantum Server causes an error. In the end this
is not required because the network is identified by its id. So
this patch removes it (as it's done in other cases as well).

Fixes bug #1187129

Change-Id: Ie18c7bd504f8c241d002d6a050b8bcc59ed9fc03

Fixed in Havana 2013.2
---
 .../dashboards/admin/networks/ports/forms.py       |    1 +
 .../dashboards/admin/networks/tests.py             |    2 --
 2 files changed, 1 insertion(+), 2 deletions(-)

diff --git a/openstack_dashboard/dashboards/admin/networks/ports/forms.py b/openstack_dashboard/dashboards/admin/networks/ports/forms.py
index 7a1f45d..939952b 100644
--- a/openstack_dashboard/dashboards/admin/networks/ports/forms.py
+++ b/openstack_dashboard/dashboards/admin/networks/ports/forms.py
@@ -57,6 +57,7 @@ class CreatePort(forms.SelfHandlingForm):
             network = api.quantum.network_get(request, data['network_id'])
             data['tenant_id'] = network.tenant_id
             data['admin_state_up'] = data['admin_state']
+            del data['network_name']
             del data['admin_state']
 
             port = api.quantum.port_create(request, **data)
diff --git a/openstack_dashboard/dashboards/admin/networks/tests.py b/openstack_dashboard/dashboards/admin/networks/tests.py
index d6c0123..9c2cd19 100644
--- a/openstack_dashboard/dashboards/admin/networks/tests.py
+++ b/openstack_dashboard/dashboards/admin/networks/tests.py
@@ -651,7 +651,6 @@ class NetworkPortTests(test.BaseAdminViewTests):
         api.quantum.port_create(IsA(http.HttpRequest),
                                 tenant_id=network.tenant_id,
                                 network_id=network.id,
-                                network_name=network.name,
                                 name=port.name,
                                 admin_state_up=port.admin_state_up,
                                 device_id=port.device_id,
@@ -688,7 +687,6 @@ class NetworkPortTests(test.BaseAdminViewTests):
         api.quantum.port_create(IsA(http.HttpRequest),
                                 tenant_id=network.tenant_id,
                                 network_id=network.id,
-                                network_name=network.name,
                                 name=port.name,
                                 admin_state_up=port.admin_state_up,
                                 device_id=port.device_id,
